### PR TITLE
Add CRaC-specific tests to GHA

### PR DESCRIPTION
**Pull request by rvansa**
Existing GitHub Actions run test tier1 but since most changes in this project focus on the CRaC capabilities we should run them in an automated fashion, too.
Right now the tests are mostly failing: this should be addressed in https://github.com/openjdk/crac/pull/47

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Anton Kozlov](https://openjdk.org/census#akozlov) (@AntonKozlov - **Reviewer**) ⚠️ Review applies to [704f50a4](https://git.openjdk.org/crac/pull/48/files/704f50a4ea9591ae1dd04aae7559492441ea4c2e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/crac.git pull/48/head:pull/48` \
`$ git checkout pull/48`

Update a local copy of the PR: \
`$ git checkout pull/48` \
`$ git pull https://git.openjdk.org/crac.git pull/48/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 48`

View PR using the GUI difftool: \
`$ git pr show -t 48`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/crac/pull/48.diff">https://git.openjdk.org/crac/pull/48.diff</a>

</details>

### CODE REVIEW

**Comment by AntonKozlov** [comment]
Looks good, but could you rebase the PR after #47 is integrated to demonstrate the GHA is green?

**Comment by rvansa** [comment]
Sure. Should I really rebase, or merge `crac` into this branch instead?

**Comment by AntonKozlov** [comment]
Please use merge. This will preserve all changes in this PR, and on the eventual integration Skara will squash all changes anyway.

**Comment by rvansa** [comment]
Apparently there's an issue with permissions, and/or the testsuite does not import correct CRIU version. I'll investigate.

**Comment by rvansa** [comment]
Closing for now to save people from notifications on push.

**Comment by rvansa** [comment]
I was finally able to resolve some test failures. Part of that problem was using an older version of CRIU (the one README links to, not release 1.3), but also ResolveTest was prone to the issue that should be fixed in #46 (added a workaround to the test).

File: test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java [hunk 1]
```diff
@@ -96,11 +98,16 @@ private static Future<?> startTestProcess() throws Exception {
         cmd.addAll(Arrays.asList("--name", CONTAINER_NAME));
         cmd.add(imageName);
         // Checkpointing does not work for PID 1, therefore we add an intermediary bash process
+        // At the same time docker --init is not sufficient; since `tini` does not wait for *all*
+        // children to finish, just the main process, creating /cr/cppath might race and we would
+        // not be able to restore the process.
+        // This workaround should not be necessary when https://github.com/openjdk/crac/pull/46
+        // is integrated.
```
Comment (AntonKozlov) [inline test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java:105]: I propose to wait until #46 is integrated and remove the mention (and workaround)

File: test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java [hunk 2]
```diff
@@ -96,11 +98,16 @@ private static Future<?> startTestProcess() throws Exception {
...
         List<String> javaCmd = new ArrayList<>();
-        javaCmd.addAll(Arrays.asList("/jdk/bin/java", "-cp /test-classes/", "-XX:CRaCCheckpointTo=/cr"));
+        javaCmd.addAll(Arrays.asList("/jdk/bin/java", "-cp", "/test-classes/", "-XX:CRaCCheckpointTo=/cr"));
```
Comment (AntonKozlov) [inline test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java:107]: This looks belonging to #50. Could this be the reason for strange failures of this test I've reported there?

File: test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java (hunk 1 above)
Comment (rvansa) [reply test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java:105]: I would integrate code when it's ready reagardless of other PRs (haven't actually checked if this works with #46). This code will be anyway rewritten with #50

File: test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java (hunk 2 above)
Comment (rvansa) [reply test/jdk/jdk/crac/java/net/InetAddress/ResolveTest.java:107]: Why would you think that? There's a different variant of this test in #50. This particular change, while correct, has no effect when we concatenate the list anyway, it's there only for clarity. I had to change the line when experimenting with `--init` version without the bash loop workaround.

**Comment by rvansa** [comment]
@AntonKozlov I've merged in recent changes and resolved the flakiness in ReseedTest. GHA now fails due to infrastructure timeout, the `crac` part of testsuite is green.

**Review by AntonKozlov — APPROVED** [review]
LGTM, thank you!

File: test/jdk/jdk/crac/SecureRandom/ReseedTest.java [hunk 3]
```diff
@@ -70,7 +71,7 @@ public void exec() throws Exception {
             throw new RuntimeException("Restore ERROR " + e);
         }
 
-        int r = sr.nextInt(255);
-        System.exit(r);
+        System.out.println(sr.nextInt());
+        System.exit(0);
```
Comment (AntonKozlov) [inline test/jdk/jdk/crac/SecureRandom/ReseedTest.java:75]: This change will also help when something is wrong with the restore, the previous version might confuse that situation with legitimate return with non-zero error. 

On the second thought, could you extract this into a separate PR?

Comment (rvansa) [reply test/jdk/jdk/crac/SecureRandom/ReseedTest.java:75]: If you promise me that you'll integrate right away, I can :)

Comment (rvansa) [reply test/jdk/jdk/crac/SecureRandom/ReseedTest.java]: Here it goes: https://github.com/openjdk/crac/pull/55
I've reverted the commit in this PR
